iothub/device/src/TransientFaultHandling/AsyncExecution.cs

```diff
@@ -88,7 +88,7 @@ private static Task<bool> StartAsGenericTask(Func<Task> taskAction)
                     nameof(taskAction));
             }
 
-            var tcs = new TaskCompletionSource<bool>();
+            TaskCompletionSource<bool> tcs = CreateTaskCompletionSource();
```

Comment:
Centralize `TaskCompletionSource` initialization to specify the preferred value (https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#always-create-taskcompletionsourcet-with-taskcreationoptionsruncontinuationsasynchronously) upon initialization.